crates/matrix-sdk-base/src/sliding_sync/mod.rs

```diff
@@ -2178,6 +2181,59 @@ mod tests {
         );
     }
 
+    #[async_test]
+    async fn test_pinned_events_are_updated_on_sync() {
+        let user_a_id = user_id!("@a:e.uk");
+        let client = logged_in_base_client(Some(user_a_id)).await;
+        let room_id = room_id!("!r:e.uk");
+        let pinned_event_id = owned_event_id!("$an-id:e.uk");
+
+        // Create room
+        let mut room_response = http::response::Room::new();
+        set_room_joined(&mut room_response, user_a_id);
+        let response = response_with_room(room_id, room_response);
+        client.process_sliding_sync(&response, &(), true).await.expect("Failed to process sync");
+
+        // The newly created room has no pinned event ids
+        let room = client.get_room(room_id).unwrap();
+        let mut pinned_events_subscriber = room.pinned_events();
+        assert!(pinned_events_subscriber.get().is_empty());
+
+        // Load new pinned event id
+        let mut room_response = http::response::Room::new();
+        room_response.required_state.push(make_state_event(
+            user_a_id,
+            "",
+            RoomPinnedEventsEventContent::new(vec![pinned_event_id.clone()]),
+            None,
+        ));
```

Comment:
https://github.com/matrix-org/matrix-rust-sdk/issues/3716 :eyes:

Reply:
I think I'll attach this in some other PR.